pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/assignment/segment/RealtimeSegmentAssignment.java

```diff
@@ -183,8 +176,10 @@ public Map<String, Map<String, String>> rebalanceTable(Map<String, Map<String, S
         config.getBoolean(RebalanceConfigConstants.BOOTSTRAP, RebalanceConfigConstants.DEFAULT_BOOTSTRAP);
 
     // Rebalance tiers first
+    SegmentAssignmentStrategy segmentAssignmentStrategy = _assignmentStrategyMap.get(InstancePartitionsType.COMPLETED);
```

Comment:
Why don't we initialize the segmentAssignmentStrategy directly from `SegmentAssignmentStrategyFactory` here?

In that case, we won't need to change the interface for `SegmentAssignment:init()` to take the `instancePartitionsMap`.

Reply:
updated